### PR TITLE
Class 06

### PR DESCRIPTION
**Pull request by irvinemd55**
Assignment work took roughly 90 minutes. Additional changes are needed based on peer review, so I'll be doing a number of reissues. Code needs to be reworked and restructured, but I was able to set up the based template to build upon.

### CODE REVIEW

File: scripts/app.js [hunk 1]
```diff
@@ -10,17 +10,22 @@ function Projects (options) {
...
-  $newProjects.find('h1').html(this.title);
-  $newProjects.find('a').html(this.author);
-  $newProjects.find('a').attr('href', this.authorUrl);
-  $newProjects.find('.projects-body').html(this.body);
-  $newProjects.find('time[pubdate]').attr('title', this.publishedOn);
-  $newProjects.find('time').text('about ' + parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000) + ' days ago');
+  // var $newProjects = $('#projects .template').clone();
+  // $newProjects.attr('data-category', this.category);
+  // $newProjects.find('h1').html(this.title);
+  // $newProjects.find('a').html(this.author);
```
Comment (patci-zz) [inline scripts/app.js:16]: You're good to remove these commented-out lines of code.

File: scripts/app.js [hunk 2]
```diff
@@ -34,3 +39,19 @@ projectObjects.forEach(function(projectsObj) {
 projects.forEach(function(projectsObj) {
   $('#projects').append(projectsObj.toHtml());
 });
+
+Article.fetchAll = function() {
+  if (localStorage.projectObjStor) {
+    var projectObjStor = JSON.parse(localStorage.getItem('projectObjStor'));
+    Article.loadAll(projectObjStor);
+    console.log('data already in localStorage');
```
Comment (patci-zz) [inline scripts/app.js:47]: If you're confident that the logic is behaving appropriately, you may also remove this console.log

File: scripts/app.js [hunk 3]
```diff
@@ -34,3 +39,19 @@ projectObjects.forEach(function(projectsObj) {
...
+    articleView.renderIndexPage();
+  } else {
+    $.getJSON('projectObjStor.json', function(data){
+      Article.loadAll(data);
+      localStorage.setItem('projectObjStor',JSON.stringify(data));
+      console.log('data not in localStorage yet');
```
Comment (patci-zz) [inline scripts/app.js:53]: Same here.

File: projectObjStor.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "body": "<p>201 Project creating a cookie stand app</p>"
+  },
+  {
+    "title": "Mobile First",
+    "category": "301",
+    "author": "Irvine Downing",
+    "authorUrl": "https://github.com/irvinemd55/01-mobile-first",
+    "publishedOn": "2016-10-24",
+    "body": "<p>301 Project modifying a website for mobile access</p>"
+  },
```
Comment (patci-zz) [inline projectObjStor.json:25]: This trailing comma might be an issue with your json

File: scripts/app.js [hunk 5]
```diff
@@ -34,3 +39,19 @@ projectObjects.forEach(function(projectsObj) {
 projects.forEach(function(projectsObj) {
   $('#projects').append(projectsObj.toHtml());
 });
+
+Article.fetchAll = function() {
```
Comment (Sanderson239) [inline scripts/app.js:43]: The fetchAll function as a method of a nonexistent Articles object so when it’s called in portfolio_view, it doesn’t do anything. Also, the loadAll and renderIndexPage functions haven’t been written yet.

File: index.html [hunk 6]
```diff
@@ -22,17 +22,19 @@ <h1 class="name">
...
-            By <address><a href="">Author Name</a></address>
-            published <time pubdate datetime="2000-01-01">Publish Time</time>
+            By <address><a href="{{authorUrl}}">{{author}}</a></address>
+            {{publishStatus}}}
                 </div>
                 </header>
               <section class="article-body"></section>
+            </script>
         </article>
       </ul>
```
Comment (Sanderson239) [inline index.html:39]: This closing </ul> tag doesn't have an opening tag.

File: scripts/portfolio_view.js [hunk 7]
```diff
@@ -13,3 +13,4 @@ portfolioView.handleMainNav = function () {
 };
 
 portfolioView.handleMainNav();
```
Comment (Sanderson239) [inline scripts/portfolio_view.js:15]: the handleMainNav function hides the tab-content, but it doesn’t show the correct content. In the index.html page, the list item for the About page needs a data-content attribute labeled after its respective section’s id which would be "about" here.

**Pull request merged**